src/Kestrel.Transport.Sockets/Internal/SocketSenderReceiverBase.cs

```diff
@@ -19,5 +20,10 @@ protected SocketSenderReceiverBase(Socket socket, PipeScheduler scheduler)
         }
 
         public void Dispose() => _awaitableEventArgs.Dispose();
+
+        protected AsyncFlowControl? SuppressExecutionContext()
+        {
+            return ExecutionContext.IsFlowSuppressed() ? (AsyncFlowControl?)null : ExecutionContext.SuppressFlow();
```

Comment:
cc @benaadams

Reply:
You can `using` a null nullable? (Tested, yes you can)

That's nice and elegant :)